test/unit/fiber.cc

```diff
@@ -742,6 +749,35 @@ fiber_test_shutdown(void)
 	header();
 }
 
+static void *
+cord_no_user_thread_rename_f(void *arg)
+{
+	(void)arg;
+#ifdef __linux__
+	const char *name = "fiber_test";
+	pthread_setname_np(pthread_self(), name);
+
+	/* Provoke creation of cord in user thread. */
+	fail_unless(strcmp(cord()->name, "unknown") == 0);
+
+	char buf[16];
+	pthread_getname_np(pthread_self(), buf, sizeof(buf));
+	fail_unless(strcmp(buf, name) == 0);
+#endif
+	return NULL;
+}
+
+static void
+cord_no_user_thread_rename(void)
+{
+	header();
+	pthread_t tid;
+	fail_if(pthread_create(&tid, NULL, cord_no_user_thread_rename_f,
+			       NULL) != 0);
+	fail_if(pthread_join(tid, NULL) != 0);
```

Comment:
Nit. Let's use `fail_unless` here too for consistency.

Reply:
Done.